### PR TITLE
CR-1124138 xbutil configure --help returns an error

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1124138
xbutil configure --help returns an error!

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Missing additional logic for top level command processing. Introduced in https://github.com/Xilinx/XRT/pull/6307

#### How problem was solved, alternative solutions (if any) and why they were rejected
Solved by adding some extra logic to check for the help menu in the top level configure command. Also fixed an issue with the device for host-mem suboption not printing an error message.

#### What has been tested and how, request additional testing if necessary
xbutil configure no options
```
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ xbutil configure
ERROR: Suboption missing

COMMAND: configure

DESCRIPTION: Device and host configuration.

USAGE: xbutil configure [-h] --[ host-mem | p2p ] [commandArgs]

OPTIONS:
  --help             - Help to use this sub-command
  --host-mem         - Controls host-mem functionality
  --p2p              - Controls P2P functionality

GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ echo $?
1
```
xbutil configure other options missing suboption
```
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ xbutil configure -d 65:00
ERROR: Suboption missing

COMMAND: configure

DESCRIPTION: Device and host configuration.

USAGE: xbutil configure [-h] --[ host-mem | p2p ] [commandArgs]

OPTIONS:
  --help             - Help to use this sub-command
  --host-mem         - Controls host-mem functionality
  --p2p              - Controls P2P functionality

GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ echo $?
1
```
xbutil configure help menu
```
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ xbutil configure --help

COMMAND: configure

DESCRIPTION: Device and host configuration.

USAGE: xbutil configure [-h] --[ host-mem | p2p ] [commandArgs]

OPTIONS:
  --help             - Help to use this sub-command
  --host-mem         - Controls host-mem functionality
  --p2p              - Controls P2P functionality

GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ echo $?
0
```
host-mem suboption device error
```
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ xbutil configure --host-mem ENABLE
ERROR: A device needs to be specified.

DESCRIPTION: Controls host-mem functionality

USAGE: xbutil configure --host-mem [--help] [-d arg] [-s arg] action
  <action>        - Action to perform: ENABLE or DISABLE

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -s, --size         - Size of host memory (bytes) to be enabled (e.g. 256M, 1G)
  --help             - Help to use this sub-command

GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
dbenusov@xsjpranjal01:/proj/xsjhdstaff6/dbenusov/XRT$ echo $?
1
```

### CODE REVIEW

File: src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp [hunk 1]
```diff
@@ -93,6 +95,7 @@ OO_HostMem::execute(const SubCmdOptions& _options) const
     throw xrt_core::error(std::errc::operation_canceled);
   }
   if(m_devices.empty()) {
+    std::cerr << boost::format("ERROR: A device needs to be specified.\n");
     printHelp();
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp:99]: I don't think we print out help when a device is not specified. For consistency purposes, please remove this. See p2p:
```
xbutil configure --p2p enable
ERROR: A device needs to be specified.
```
For the future, if a device is not specified, we should show the error message and optionally show the list of available devices, for eg:
```
$ xbutil validate 

ERROR: Please specify a single device using --device option

List of available devices:
  [0000:b3:00.1] : xilinx_u200_gen3x16_xdma_base_1
  [0000:18:00.1] : xilinx_u30_gen3x4_base_2
  [0000:17:00.1] : xilinx_u30_gen3x4_base_2

```

**Review by AShivangi — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp (hunk 1 above)
Comment (dbenusov-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp:99]: Fixed!

**Review by AShivangi — APPROVED** [review]

File: src/runtime_src/core/tools/xbutil2/SubCmdConfigure.cpp [hunk 2]
```diff
@@ -119,9 +119,14 @@ SubCmdConfigure::execute(const SubCmdOptions& _options) const
 
   // No suboption print help
   if (!optionOption) {
-    std::cerr << "ERROR: Suboption missing" << std::endl;
-    printHelp(commonOptions, hiddenOptions, subOptionOptions);
-    throw xrt_core::error(std::errc::operation_canceled);
+    if (help) {
+      printHelp(commonOptions, hiddenOptions, subOptionOptions);
+      return;
+    } else {
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdConfigure.cpp:125]: `else `statement is not needed because of the `return `on line 124.

**Review by rozumx — CHANGES_REQUESTED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please.

**Review by rozumx — APPROVED** [review]

**Pull request merged**